lib/ecco/save_event_listener.rb

```diff
@@ -2,30 +2,31 @@
 
 module Ecco
   class SaveEventListener < EventListener
-    SAVE_EVENTS = [
-      EventType::QUERY,
-      EventType::ROTATE,
-      EventType::WRITE_ROWS,
-      EventType::EXT_WRITE_ROWS,
-      EventType::UPDATE_ROWS,
-      EventType::EXT_UPDATE_ROWS,
-      EventType::DELETE_ROWS,
-      EventType::EXT_DELETE_ROWS,
-    ]
+    SAVE_EVENTS = {
+      EventType::QUERY           => "QUERY",
+      EventType::ROTATE          => "ROTATE",
+      EventType::WRITE_ROWS      => "WRITE_ROWS",
+      EventType::EXT_WRITE_ROWS  => "WRITE_ROWS",
+      EventType::UPDATE_ROWS     => "UPDATE_ROWS",
+      EventType::EXT_UPDATE_ROWS => "UPDATE_ROWS",
+      EventType::DELETE_ROWS     => "DELETE_ROWS",
+      EventType::EXT_DELETE_ROWS => "DELETE_ROWS",
+    }
 
     def accepted_events
-      SAVE_EVENTS
+      SAVE_EVENTS.keys
     end
 
     def on_event(event)
       type = event.get_header.get_event_type
 
       case type
       when *accepted_events
+        event_type_name = SAVE_EVENTS[type]
         filename = @client.get_binlog_filename
         position = @client.get_binlog_position
 
-        @callback.call(filename, position)
+        @callback.call(event_type_name, filename, position)
```

Comment:
I think `event_type_name` should be the last argument, since

* It's the least important, the main reason for this event is to persist the position, thus filename and position is more important.
* Keeps compatibility with old code: `Proc.new { |filename, position| # Will still work as before }`.
* Most likely to be ignore `Proc.new { |filename, position, _| # Save on all events }`.